File: android/src/main/java/com/oblador/keychain/exceptions/CryptoFailedException.java
package com.oblador.keychain.exceptions;

import androidx.annotation.Nullable;

import java.security.GeneralSecurityException;

public class CryptoFailedException extends GeneralSecurityException {
  public CryptoFailedException(String message) {
    super(message);
  }

  public CryptoFailedException(String message, Throwable t) {
    super(message, t);
  }

  public static void reThrowOnError(@Nullable final Throwable error) throws CryptoFailedException {
    if(null == error) return;

    if (error instanceof CryptoFailedException)
      throw (CryptoFailedException) error;

    throw new CryptoFailedException("Wrapped error: " + error.getMessage(), error);

  }
}

File: android/src/main/java/com/oblador/keychain/exceptions/CryptoFailedException.kt
package com.oblador.keychain.exceptions

import java.security.GeneralSecurityException

class CryptoFailedException : GeneralSecurityException {
  constructor(message: String?) : super(message)

  constructor(message: String?, t: Throwable?) : super(message, t)

  companion object {
    @JvmStatic
    @Throws(CryptoFailedException::class)
    fun reThrowOnError(error: Throwable?) {
      if (null == error) return
      if (error is CryptoFailedException) throw (error as CryptoFailedException?)!!
      throw CryptoFailedException("Wrapped error: " + error.message, error)
    }
  }
}
